/*
 * SPDX-License-Identifier: MIT
 * SPDX-FileCopyrightText: Copyright (c) 2024 OpenStickCommunity (gp2040-ce.info)
 */

#ifndef PICO_BOARD_CONFIG_H_
#define PICO_BOARD_CONFIG_H_

#include "enums.pb.h"
#include "class/hid/hid.h"

#define BOARD_CONFIG_LABEL "HML"

// Main pin mapping Configuration
//                                                  // GP2040 | Xinput | Switch  | PS3/4/5  | Dinput | Arcade |
#define GPIO_PIN_13 GpioAction::BUTTON_PRESS_UP     // UP     | UP     | UP      | UP       | UP     | UP     |
#define GPIO_PIN_12 GpioAction::BUTTON_PRESS_DOWN   // DOWN   | DOWN   | DOWN    | DOWN     | DOWN   | DOWN   |
#define GPIO_PIN_03 GpioAction::BUTTON_PRESS_LEFT   // LEFT   | LEFT   | LEFT    | LEFT     | LEFT   | LEFT   |
#define GPIO_PIN_02 GpioAction::BUTTON_PRESS_RIGHT  // RIGHT  | RIGHT  | RIGHT   | RIGHT    | RIGHT  | RIGHT  |
#define GPIO_PIN_04 GpioAction::BUTTON_PRESS_B1     // B1     | A      | B       | Cross    | 2      | K1     |
#define GPIO_PIN_10 GpioAction::BUTTON_PRESS_B2     // B2     | B      | A       | Circle   | 3      | K2     |
#define GPIO_PIN_07 GpioAction::BUTTON_PRESS_R2     // R2     | RT     | ZR      | R2       | 8      | K3     |
#define GPIO_PIN_18 GpioAction::BUTTON_PRESS_L2     // L2     | LT     | ZL      | L2       | 7      | K4     |
#define GPIO_PIN_09 GpioAction::BUTTON_PRESS_B3     // B3     | X      | Y       | Square   | 1      | P1     |
#define GPIO_PIN_05 GpioAction::BUTTON_PRESS_B4     // B4     | Y      | X       | Triangle | 4      | P2     |
#define GPIO_PIN_06 GpioAction::BUTTON_PRESS_R1     // R1     | RB     | R       | R1       | 6      | P3     |
#define GPIO_PIN_19 GpioAction::BUTTON_PRESS_L1     // L1     | LB     | L       | L1       | 5      | P4     |
#define GPIO_PIN_00 GpioAction::BUTTON_PRESS_S1     // S1     | Back   | Minus   | Select   | 9      | Coin   |
#define GPIO_PIN_01 GpioAction::BUTTON_PRESS_S2     // S2     | Start  | Plus    | Start    | 10     | Start  |
#define GPIO_PIN_22 GpioAction::BUTTON_PRESS_L3     // L3     | LS     | LS      | L3       | 11     | LS     |
#define GPIO_PIN_24 GpioAction::BUTTON_PRESS_R3     // R3     | RS     | RS      | R3       | 12     | RS     |
#define GPIO_PIN_11 GpioAction::BUTTON_PRESS_A1     // A1     | Guide  | Home    | PS       | 13     | ~      |
#define GPIO_PIN_08 GpioAction::BUTTON_PRESS_A2     // A2     | ~      | Capture | ~        | 14     | ~      |
#define GPIO_PIN_17 GpioAction::BUTTON_PRESS_L3     // L3     | LS     | LS      | L3       | 11     | LS     | 
#define GPIO_PIN_25 GpioAction::BUTTON_PRESS_R3     // R3     | RS     | RS      | R3       | 12     | RS     |  

// Setting GPIO pins to assigned by add-on
//
#define GPIO_PIN_16 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_20 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_21 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_26 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_27 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_28 GpioAction::ASSIGNED_TO_ADDON
#define GPIO_PIN_29 GpioAction::ASSIGNED_TO_ADDON

// Keyboard Mapping Configuration
//                                            // GP2040 | Xinput | Switch  | PS3/4/5  | Dinput | Arcade |
#define KEY_DPAD_UP     HID_KEY_ARROW_UP      // UP     | UP     | UP      | UP       | UP     | UP     |
#define KEY_DPAD_DOWN   HID_KEY_ARROW_DOWN    // DOWN   | DOWN   | DOWN    | DOWN     | DOWN   | DOWN   |
#define KEY_DPAD_RIGHT  HID_KEY_ARROW_RIGHT   // RIGHT  | RIGHT  | RIGHT   | RIGHT    | RIGHT  | RIGHT  |
#define KEY_DPAD_LEFT   HID_KEY_ARROW_LEFT    // LEFT   | LEFT   | LEFT    | LEFT     | LEFT   | LEFT   |
#define KEY_BUTTON_B1   HID_KEY_SHIFT_LEFT    // B1     | A      | B       | Cross    | 2      | K1     |
#define KEY_BUTTON_B2   HID_KEY_Z             // B2     | B      | A       | Circle   | 3      | K2     |
#define KEY_BUTTON_R2   HID_KEY_X             // R2     | RT     | ZR      | R2       | 8      | K3     |
#define KEY_BUTTON_L2   HID_KEY_V             // L2     | LT     | ZL      | L2       | 7      | K4     |
#define KEY_BUTTON_B3   HID_KEY_CONTROL_LEFT  // B3     | X      | Y       | Square   | 1      | P1     |
#define KEY_BUTTON_B4   HID_KEY_ALT_LEFT      // B4     | Y      | X       | Triangle | 4      | P2     |
#define KEY_BUTTON_R1   HID_KEY_SPACE         // R1     | RB     | R       | R1       | 6      | P3     |
#define KEY_BUTTON_L1   HID_KEY_C             // L1     | LB     | L       | L1       | 5      | P4     |
#define KEY_BUTTON_S1   HID_KEY_5             // S1     | Back   | Minus   | Select   | 9      | Coin   |
#define KEY_BUTTON_S2   HID_KEY_1             // S2     | Start  | Plus    | Start    | 10     | Start  |
#define KEY_BUTTON_L3   HID_KEY_EQUAL         // L3     | LS     | LS      | L3       | 11     | LS     |
#define KEY_BUTTON_R3   HID_KEY_MINUS         // R3     | RS     | RS      | R3       | 12     | RS     |
#define KEY_BUTTON_A1   HID_KEY_9             // A1     | Guide  | Home    | PS       | 13     | ~      |
#define KEY_BUTTON_A2   HID_KEY_F2            // A2     | ~      | Capture | ~        | 14     | ~      |
#define KEY_BUTTON_FN   -1                    // Hotkey Function                                        |

// LED Configuration
#define BOARD_LEDS_PIN 16
#define LED_BRIGHTNESS_MAXIMUM 80
#define CASE_RGB_TYPE  CASE_RGB_TYPE_AMBIENT
#define CASE_RGB_INDEX 0
#define CASE_RGB_COUNT 1

// analog stick configuration
#define ANALOG_INPUT_ENABLED        1
#define ANALOG_ADC_1_VRX            27
#define ANALOG_ADC_1_VRY            26
#define ANALOG_ADC_2_VRX            29
#define ANALOG_ADC_2_VRY            28
#define FORCED_CIRCULARITY_ENABLED  1
#define FORCED_CIRCULARITY2_ENABLED 1
#define DEFAULT_INNER_DEADZONE      0
#define DEFAULT_INNER_DEADZONE2     0
#define DEFAULT_OUTER_DEADZONE      63
#define DEFAULT_OUTER_DEADZONE2     63
#define AUTO_CALIBRATE_ENABLED      0
#define AUTO_CALIBRATE2_ENABLED     0
#define ANALOG_ERROR 958
#define ANALOG_ERROR2 958

// USB Passthrough
//#define USB_PERIPHERAL_ENABLED 1
//#define USB_PERIPHERAL_PIN_DPLUS 14
//#define USB_PERIPHERAL_PIN_ORDER 0

// PS AUTH
#define DEFAULT_PS4CONTROLLER_TYPE PS4_CONTROLLER
#define DEFAULT_PS4_ID_MODE PS4_ID_EMULATION
#define DEFAULT_PS4AUTHENTICATION_TYPE INPUT_MODE_AUTH_TYPE_KEYS

// Input Modes
#define DEFAULT_INPUT_MODE INPUT_MODE_PS4
#define DEFAULT_INPUT_MODE_B2 INPUT_MODE_XINPUT
#define DEFAULT_INPUT_MODE_B3 INPUT_MODE_PS4
#define DEFAULT_INPUT_MODE_B4 -1
#define DEFAULT_INPUT_MODE_B1 -1
#define DEFAULT_INPUT_MODE_R2 -1
#define DEFAULT_INPUT_MODE_L2 -1
#define DEFAULT_INPUT_MODE_R1 -1
#define DEFAULT_INPUT_MODE_L1 -1

// mini led
#define MINI_MENU_GAMEPAD_INPUT 1 
#define HAS_I2C_DISPLAY 1
#define I2C0_ENABLED 1
#define I2C0_PIN_SDA 20
#define I2C0_PIN_SCL 21
#define SPLASH_MODE SPLASH_MODE_STATIC
#define SPLASH_DURATION 2000

// MINI LED layout
 #define BUTTON_LAYOUT BUTTON_LAYOUT_BOARD_DEFINED_A
 #define BUTTON_LAYOUT_RIGHT BUTTON_LAYOUT_BOARD_DEFINED_B

 // HML layout WASD
 #define DEFAULT_BOARD_LAYOUT_A {\
    {GP_ELEMENT_PIN_BUTTON, {19, 26, 26, 33, 1, 1,  5,     GP_SHAPE_SQUARE, 22}},\
    {GP_ELEMENT_PIN_BUTTON, {27, 30, 34, 37, 1, 1,  3,     GP_SHAPE_SQUARE, 22}},\
    {GP_ELEMENT_PIN_BUTTON, {35, 34, 42, 41, 1, 1,  4,     GP_SHAPE_SQUARE, 22}},\
    {GP_ELEMENT_PIN_BUTTON, {27, 22, 42, 29, 1, 1, 18,     GP_SHAPE_SQUARE, 22}}\
}

// HML layout right
#define DEFAULT_BOARD_LAYOUT_B {\
    {GP_ELEMENT_BTN_BUTTON, {96, 30, 12, 12, 1, 1, GAMEPAD_MASK_B4,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {79, 42, 12, 12, 1, 1, GAMEPAD_MASK_B3,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {113, 42, 12, 12, 1, 1, GAMEPAD_MASK_B2,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {96, 55, 12, 12, 1, 1, GAMEPAD_MASK_B1,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {79,  9, 15, 7, 1, 1, GAMEPAD_MASK_L1,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {109, 9, 15, 7, 1, 1, GAMEPAD_MASK_R1,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {79,  17, 15, 7, 1, 1, GAMEPAD_MASK_L2,      GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_BTN_BUTTON, {109, 17, 15, 7, 1, 1, GAMEPAD_MASK_R2,      GP_SHAPE_ELLIPSE}}\
}

// HML logo
#define DEFAULT_SPLASH \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x3C, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x3F, 0xF8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x3F, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3F, 0x00, 0x00, \
  0x3F, 0xFF, 0xA3, 0xE0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0xFF, 0x40, 0x00, \
  0x3F, 0xFF, 0xB3, 0xFF, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0x60, 0x00, \
  0x3F, 0xFF, 0xB3, 0xFF, 0xF8, 0x07, 0xFF, 0xC0, 0x00, 0xFF, 0xFE, 0x00, 0xFF, 0xFF, 0x60, 0x00, \
  0x1F, 0xFF, 0xB3, 0xFF, 0xFB, 0x07, 0xFF, 0xF0, 0x01, 0xFF, 0xFE, 0x00, 0xFF, 0xFF, 0x60, 0x00, \
  0x01, 0xF8, 0x33, 0xFF, 0xFB, 0x07, 0xFF, 0xFA, 0x01, 0xFF, 0xFE, 0x40, 0xFF, 0xFF, 0x70, 0x00, \
  0x01, 0xF8, 0x30, 0x7F, 0xFB, 0x87, 0xFF, 0xF9, 0x01, 0xFF, 0xFE, 0x40, 0xFF, 0xE0, 0x70, 0x00, \
  0x19, 0xF9, 0xF0, 0x3F, 0x83, 0x87, 0xFF, 0xF9, 0x83, 0xFF, 0xFE, 0x60, 0x0F, 0xE0, 0x70, 0x00, \
  0x09, 0xFB, 0xF1, 0x3F, 0x03, 0x80, 0x7F, 0xFC, 0xC3, 0xFF, 0xC0, 0x60, 0x07, 0xEF, 0xF0, 0x00, \
  0x09, 0xF9, 0xF1, 0x3F, 0x3F, 0x83, 0x7F, 0xFC, 0xC3, 0xFF, 0xCF, 0xE0, 0x67, 0xEF, 0xF0, 0x00, \
  0x01, 0xF9, 0xF0, 0x3F, 0x3F, 0x83, 0x7E, 0xFE, 0xE7, 0xEF, 0xCF, 0xE0, 0x27, 0xEF, 0xF0, 0x00, \
  0x01, 0xF9, 0xF0, 0x3F, 0x3F, 0x81, 0x7E, 0xFE, 0x67, 0xEF, 0xCF, 0xE0, 0x27, 0xEF, 0xE0, 0x00, \
  0x01, 0xF9, 0x80, 0x3F, 0x3F, 0x00, 0x7E, 0x7E, 0x67, 0xCF, 0xCF, 0xE0, 0x07, 0xEE, 0x00, 0x00, \
  0x01, 0xF8, 0x00, 0x3F, 0x30, 0x00, 0x7E, 0x7F, 0x6F, 0xCF, 0xCC, 0x00, 0x07, 0xEE, 0x00, 0x00, \
  0x01, 0xFF, 0xE0, 0x3F, 0x30, 0x00, 0x7E, 0x7F, 0x0F, 0xCF, 0xCC, 0x00, 0x07, 0xEE, 0x00, 0x00, \
  0x01, 0xFF, 0xFF, 0xFF, 0x30, 0x00, 0x7E, 0x3F, 0x8F, 0x8F, 0xCC, 0x00, 0x07, 0xEE, 0x00, 0x00, \
  0x01, 0xFF, 0xFF, 0xFF, 0x30, 0x00, 0x7E, 0x3F, 0x9F, 0x8F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xC0, \
  0x01, 0xFF, 0xFF, 0xFF, 0x30, 0x00, 0x7E, 0x9F, 0x9F, 0x8F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xC8, \
  0x01, 0xFF, 0xFF, 0xFF, 0x30, 0x00, 0x7E, 0x9F, 0xFF, 0x0F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF8, 0x00, 0x3F, 0x30, 0x00, 0x7E, 0xDF, 0xFF, 0x6F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF9, 0xFF, 0x3F, 0x30, 0x00, 0x7E, 0xCF, 0xFE, 0x6F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF9, 0xFF, 0x3F, 0x30, 0x00, 0x7E, 0xEF, 0xFE, 0x6F, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF9, 0xFF, 0x3F, 0x30, 0x00, 0x7E, 0xE7, 0xFE, 0xEF, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF9, 0xFF, 0x3F, 0x30, 0x00, 0x7E, 0xE7, 0xFC, 0xEF, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF9, 0x80, 0x3F, 0x30, 0x00, 0x7E, 0x07, 0xFC, 0xCF, 0xCC, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF8, 0x00, 0x3F, 0x00, 0x00, 0x7E, 0x03, 0xFC, 0x0F, 0xC0, 0x00, 0x07, 0xEE, 0x0F, 0xCC, \
  0x01, 0xF8, 0x00, 0x7F, 0xF8, 0x07, 0xFF, 0xC3, 0xF8, 0x7F, 0xFE, 0x00, 0x0F, 0xE0, 0x0F, 0xCC, \
  0x01, 0xFF, 0x03, 0xFF, 0xF8, 0x07, 0xFF, 0xCB, 0xFA, 0x7F, 0xFE, 0x00, 0xFF, 0xF8, 0x0F, 0xCC, \
  0x3F, 0xFF, 0x83, 0xFF, 0xFB, 0x07, 0xFF, 0xC9, 0xFA, 0x7F, 0xFE, 0x40, 0xFF, 0xFF, 0xFF, 0xCC, \
  0x3F, 0xFF, 0xB3, 0xFF, 0xFB, 0x87, 0xFF, 0xCD, 0xF2, 0x7F, 0xFE, 0x60, 0xFF, 0xFF, 0xFF, 0xCC, \
  0x3F, 0xFF, 0xBB, 0xFF, 0xFB, 0x87, 0xFF, 0xCC, 0x76, 0x7F, 0xFE, 0x60, 0xFF, 0xFF, 0xFF, 0xCC, \
  0x3F, 0xFF, 0xBB, 0xFF, 0x83, 0x80, 0x00, 0x0C, 0x06, 0x00, 0x00, 0x60, 0xFF, 0xFF, 0xFF, 0xCC, \
  0x3F, 0xFF, 0x38, 0x00, 0x03, 0x83, 0xFF, 0xFC, 0x7E, 0x00, 0x00, 0x60, 0x0F, 0xFF, 0xFF, 0xCC, \
  0x3F, 0xC0, 0x38, 0x0F, 0xFF, 0x83, 0xFF, 0xFC, 0x3E, 0x3F, 0xFF, 0xE0, 0x00, 0x3F, 0xFF, 0xCC, \
  0x10, 0x00, 0x79, 0xFF, 0xFF, 0x81, 0xFF, 0xFC, 0x1E, 0x1F, 0xFF, 0xE0, 0x7C, 0x00, 0x0F, 0xCC, \
  0x00, 0x1F, 0xF8, 0xFF, 0xFF, 0x80, 0xFF, 0xF0, 0x0C, 0x1F, 0xFF, 0xE0, 0x3F, 0xF0, 0x00, 0x0C, \
  0x07, 0xFF, 0xF8, 0x7F, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1F, 0xE0, 0x3F, 0xFF, 0x80, 0x0C, \
  0x1F, 0xFF, 0xF8, 0x78, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1F, 0xFF, 0xFF, 0xFC, \
  0x0F, 0xFF, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xFF, 0xFF, 0xFC, \
  0x0F, 0xFF, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0xFF, 0xFC, \
  0x07, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3F, 0xFC, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xFC, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00

#endif
